### PR TITLE
changing to w3 for gas price, estimate, nonce, create raw

### LINKED ISSUE #5: Migrate all supported RPC calls from ethclient to w3 - https://github.com/lmittmann/w3

### PR DESCRIPTION
**Pull request by krishnateja262**
@Jesserc this is a quick WIP PR, let me know if this aligns to what you are looking for. If its fine i can continue working on the remaining files

### CODE REVIEW

**Review by Jesserc — APPROVED** [review]
This is great🔥

**Comment by Jesserc** [comment]
@krishnateja262 did you run the test?

**Comment by krishnateja262** [comment]
@Jesserc yes, the test cases passed for the all the committed files in this PR, i have almost completed making changes in other files as well will push them soon, `handleSend.go` success cases were failing, in main branch as well, trying to fix this

**Comment by krishnateja262** [comment]
@Jesserc i have finished for migration for all the files. Can you check and let me know if any further changes required

File: cmd/tx/handleCreateContract.go [hunk 1]
```diff
@@ -43,35 +47,37 @@ func CreateContract(rpcURL, data, privateKey string, gasLimit, wei uint64) (*TxR
...
-	chainID, err := client.ChainID(context.Background())
-	if err != nil {
-		return nil, fmt.Errorf("failed to get chain: %s", err)
-	}
-
-	// Get base fee
-	baseFee, err := client.SuggestGasPrice(context.Background())
-	if err != nil {
-		return nil, fmt.Errorf("failed to get base fee: %s", err)
+	var chainID uint64
```
Comment (Jesserc) [inline cmd/tx/handleCreateContract.go:56]: Can these variables be put inside a single var block?

File: cmd/tx/handleCreateContract.go [hunk 2]
```diff
@@ -120,27 +134,30 @@ func CreateContract(rpcURL, data, privateKey string, gasLimit, wei uint64) (*TxR
...
 	}
 
-	signedTx, err := types.SignTx(tx, types.LatestSignerForChainID(chainID), ecdsaPrivateKey)
+	signedTx, err := types.SignTx(tx, types.LatestSignerForChainID(big.NewInt(int64(chainID))), ecdsaPrivateKey)
 	if err != nil {
 		return nil, fmt.Errorf("failed to sign transaction: %s", err)
 	}
 
-	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	ctx, cancel := context.WithTimeout(context.Background(), 35*time.Second)
```
Comment (Jesserc) [inline cmd/tx/handleCreateContract.go:142]: any reason for increasing the timeout to 35s?

File: cmd/tx/handleCreateContract.go (hunk 1 above)
Comment (Jesserc) [inline cmd/tx/handleCreateContract.go]: Can these variables be put inside a single var block?

'chainID, baseFee, priorityFee, errs'

File: cmd/tx/handleSendRaw.go [hunk 3]
```diff
@@ -60,18 +63,27 @@ func SendRawTransaction(rawTx, rpcURL string) (string, string, error) {
...
 	log.Warn("Sending transaction, please wait for confirmation...")
 
-	if err = client.SendTransaction(context.Background(), tx); err != nil {
-		return "", "", fmt.Errorf("failed to send transaction: %s", err)
-	}
-	// Get chain ID
-	chainID, err := client.ChainID(context.Background())
-	if err != nil {
-		return "", "", fmt.Errorf("failed to get chain ID: %s", err)
+	var res common.Hash
```
Comment (Jesserc) [inline cmd/tx/handleSendRaw.go]: can we fit all the vars into a single var() block

File: cmd/tx/handleSendBlobTx.go [hunk 4]
```diff
@@ -82,13 +81,30 @@ func SendBlobTX(rpcURL, toAddress, data, privateKey, saveBlobDir string) (string
 
 	fromAddress := crypto.PubkeyToAddress(ecdsaPrivateKey.PublicKey)
 
-	nonce, err := client.PendingNonceAt(context.Background(), fromAddress)
+	var chainID uint64
```
Comment (Jesserc) [inline cmd/tx/handleSendBlobTx.go]: can we group the vars here too

**Comment by Jesserc** [comment]
@krishnateja262 everything looks good, just a few comments to address
please can you share a screenshot of the test passing when you're done? thx

**Comment by krishnateja262** [comment]
@Jesserc Done. PFA for the screenshots, i have attached both verbose and non verbose


Screenshot 2024-06-06 at 12 57 16 AM
Screenshot 2024-06-06 at 1 03 05 AM
Screenshot 2024-06-06 at 1 02 45 AM
Screenshot 2024-06-06 at 1 02 45 AM
Screenshot 2024-06-06 at 1 02 25 AM
Screenshot 2024-06-06 at 1 02 25 AM
6b1-b71b-3725f51edd92">
Screenshot 2024-06-06 at 12 57 16 AM

**Comment by Jesserc** [comment]
@krishnateja262 thank you for taking the time to do this🙌🎉

**Review by Jesserc — APPROVED** [review]

**Comment by krishnateja262** [comment]
@Jesserc it was my pleasure :)

**Pull request merged**